### PR TITLE
added ibm monthly invoice to  generate tag reports

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: cloud_governance/common/clouds/ibm/account/ibm_account.py [hunk 1]
```diff
@@ -43,26 +46,111 @@ def __organise_user_tags(self, tags: dict):
                 user_tags.append(f'{tag.strip().lower()}:{value.strip().lower()}')
         return user_tags
 
-    def get_user_tags_from_gsheet(self, username: str, file_path: str = '/tmp/'):
+    def get_user_tags_from_gsheet(self, username: str, user_email: str, file_path: str = '/tmp/'):
```
Comment (ebattat) [inline cloud_governance/common/clouds/ibm/account/ibm_account.py:49]: Need to add @typechecked()

File: cloud_governance/common/clouds/ibm/account/ibm_account.py [hunk 2]
```diff
@@ -13,10 +13,13 @@ class IBMAccount:
...
     """
 
+    START_DATE = 1
+    END_DATE = 16
+
     def __init__(self):
         self.__API_USERNAME = os.environ.get('IBM_API_USERNAME', '')
         self.__API_KEY = os.environ.get('IBM_API_KEY', '')
-        self.__sl_client = SoftLayer.Client(username=self.__API_USERNAME, api_key=self.__API_KEY)
+        self.__sl_client = SoftLayer.create_client_from_env(username=self.__API_USERNAME, api_key=self.__API_KEY)
```
Comment (ebattat) [inline cloud_governance/common/clouds/ibm/account/ibm_account.py:22]: add try catch

File: cloud_governance/policy/ibm/ibm_cost_report.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    @typechecked
+    def beautify_and_aggregate_data(self, resource_data: dict, invoice_data: dict):
+        """
+        This method concatenate two dicts and aggregate its sum
+        @param resource_data:
+        @param invoice_data:
+        @return:
+        """
+        pd1 = pandas.DataFrame(list(resource_data.values()))
```
Comment (ebattat) [inline cloud_governance/policy/ibm/ibm_cost_report.py:96]: You should use meaningful name and not pd1, pd2
maybe invoice_df, resource_df

Comment (athiruma) [reply cloud_governance/policy/ibm/ibm_cost_report.py:96]: ok

File: cloud_governance/policy/ibm/ibm_cost_report.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        for fqdn in invoice_data.keys():
+            if fqdn not in collect_machines_data:
+                user = [user for user in users if user in fqdn]
+                if user:
+                    user_tags = self.ibm_account.get_user_tags_from_gsheet(username=f'{user[0]}@redhat.com', user_email='yes')
+                    invoice_data[fqdn].update(self.collect_tags_from_machines(user_tags))
+        return invoice_data
+
+    @typechecked
+    def beautify_and_aggregate_data(self, resource_data: dict, invoice_data: dict):
```
Comment (ebattat) [inline cloud_governance/policy/ibm/ibm_cost_report.py:89]: beautify_and_aggregate_data -> concatenate dictionaries

File: cloud_governance/policy/ibm/ibm_cost_report.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        This method concatenate two dicts and aggregate its sum
+        @param resource_data:
+        @param invoice_data:
+        @return:
+        """
+        resource_pd = pandas.DataFrame(list(resource_data.values()))
```
Comment (ebattat) [inline cloud_governance/policy/ibm/ibm_cost_report.py:96]: df not pd

Comment (athiruma) [reply cloud_governance/policy/ibm/ibm_cost_report.py:96]: pandas_dataframe

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**